test/python/test_qobj.py

```diff
@@ -39,7 +40,46 @@ def test_create_qobj(self):
                     {'name': 'u1', 'params': [0.4], 'qubits': [1]},
                     {'name': 'u2', 'params': [0.4, 0.2], 'qubits': [1]}
                 ]}
-            ]
+            ],
         }
 
-        self.assertEqual(qobj.as_dict(), expected)
+    def test_as_dict(self):
+        """Test conversion to dict of a Qobj based on the individual elements."""
+        self.valid_qobj._version = '67890'  # private member variables shouldn't appear in the dict
+        self.assertEqual(self.valid_qobj.as_dict(), self.valid_dict)
+
+    def test_as_dict_against_schema(self):
+        """Test dictionary representation of Qobj against its schema."""
+        file_path = self._get_resource_path('qobj_schema.json', Path.SCHEMAS)
+
+        with open(file_path, 'r') as schema_file:
+            schema = json.load(schema_file)
+
+        try:
+            jsonschema.validate(self.valid_qobj.as_dict(), schema)
+        except jsonschema.ValidationError as validation_error:
+            self.fail(str(validation_error))
+
+    @unittest.expectedFailure  # expected to fail until _qobjectify_item is updated
+    def test_from_dict_per_class(self):
+        """Test Qobj and its subclass representations given a dictionary."""
+        qobj_config_dict = {'shots': 1, 'memory_slots': 2}
+        qobj_experiment_dict = {'instructions':
+                                [QobjInstruction(name='u1', qubits=[1], params=[0.4])]}
+        qobj_instruction_dict = {'name': 'u1', 'qubits': [1], 'params': [0.4]}
+
+        valid_qobj_config = QobjConfig(shots=1024, memory_slots=2, max_credits=10)
+        valid_qobj_experiment = QobjExperiment(
+            instructions=[QobjInstruction(name='u1', qubits=[1], params=[0.4])]
+        )
+        valid_qobj_instruction = QobjInstruction(name='u1', qubits=[1], params=[0.4])
+
+        dicts = [self.valid_dict, qobj_config_dict, qobj_experiment_dict, qobj_instruction_dict]
+        objects = [self.valid_qobj, valid_qobj_config,
+                   valid_qobj_experiment, valid_qobj_instruction]
+        classes = [Qobj, QobjConfig, QobjExperiment, QobjInstruction]
+
+        for objekt, dikt, klass in zip(objects, dicts, classes):
```

Comment:
Small nit - can you revise the names of these variables? From [PEP 8](https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments):

> If a function argument's name clashes with a reserved keyword, it is generally better to append a single trailing underscore rather than use an abbreviation or spelling corruption. Thus class_ is better than clss

That paragraph is related to function argument's names, but applies in general too.

Reply:
I completely agree with Diego but try to come up with better names. You have `Qobj` instances (not `object` instances), `QItem` derived classes and expected dictionaries. If you can not come up with meaningful names, adhere to Diego's recommendation.

Reply:
I completely agree with Diego but try to come up with better names. You have `Qobj` instances (not `object` instances), `QItem` derived classes and expected dictionaries. If you can not come up with meaningful names, adhere to Diego's recommendation.